controllers/users.js

```diff
@@ -13,46 +19,105 @@ module.exports.getUser = (req, res, next) => {
   User.find({ _id: req.params.id })
     .then((user) => {
       if (!user || !user.length) {
-        res.status(404).send({ message: 'Нет пользователя с таким id' });
+        throw new NotFoundError('Нет пользователя с таким id');
+        // res.status(404).send({ message: 'Нет пользователя с таким id' });
       } else {
         res.send(user);
       }
     })
     .catch((err) => {
-      console.log('Что-то пошло не так при загрузке пользователя. ', err);
       next(err);
     });
 };
 
 module.exports.createUser = (req, res, next) => {
-  const { name, about, avatar } = req.body;
+  const {
+    name,
+    about,
+    avatar,
+    email,
+    password,
+  } = req.body;
 
-  User.create({ name, about, avatar })
-    .then((user) => res.send({ data: user }))
+  bcrypt.hash(password, 10)
+    .then((hash) => User.create({
+      name,
+      about,
+      avatar,
+      email,
+      password: hash,
+    }))
+    .then((user) => res.send(
+      {
+        data: {
+          name: user.name,
+          about: user.about,
+          avatar: user.avatar,
+          email: user.email,
+        },
+      },
+    ))
     .catch((err) => {
-      console.log('Что-то пошло не так при создании пользователя. ', err);
-      next(err);
+      if (err.code === 11000) {
+        throw new AlreadyExistError(err.errmsg);
```

Comment:
Нужно исправить: при попытке повторной регистрации сервер не отвечает.
В консоли ошибка:
```bash
(node:11407) UnhandledPromiseRejectionWarning: Error: E11000 duplicate key error collection: mestodb.users index: email_1 dup key: { email: "petya7@example.com" }
    at bcrypt.hash.then.then.catch (/home/jmi/work/ya-reviews/mesto-server/controllers/users.js:62:15)
    at process._tickCallback (internal/process/next_tick.js:68:7)
(node:11407) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 3)
```

Это происходит из-за того, что ошибка не попадает в обрабочик ошибок.
```suggestion
        next(new AlreadyExistError(err.errmsg));
```